src/pages/settings/panes/Languages.tsx

```diff
@@ -53,6 +53,24 @@ export function Component(props: Props) {
         Langs[x as keyof typeof Langs],
     ]) as Key[];
 
+    // Get the user's system language
+    const preferredLanguage =
+        navigator.languages
+            ?.filter((lang) =>
+                languages.find((l) => l[0] == lang.split("-")[0]),
+            )?.[0]
+            ?.split("-")[0] ?? "en";
+
+    // This moves the user's system language to the top of the language list
+    const prefLangKey = languages.find((lang) => lang[0] == preferredLanguage);
```

Comment:
Remove `?? "en"` to avoid having to splice English from the first position to the first position since it's already there by default.

Could wrap L65 onwards with:
```javascript
if (preferredLanguage) {
```

Reply:
Done.